alts/src/main/java/io/grpc/alts/internal/AltsProtocolNegotiator.java

```diff
@@ -91,13 +101,34 @@ public Handler newHandler(GrpcHttp2ConnectionHandler grpcHandler) {
       @Override
       public void close() {
         logger.finest("ALTS Server ProtocolNegotiator Closed");
-        // TODO(jiangtaoli2016): release resources
+        handshakerChannelPool.returnObject(lazyHandshakerChannel.get());
```

Comment:
This creates it if it wasn't used. If that's being done there's no point in creating it lazily.

Add a close/shutdown/release to `LazyChannel`? It would be a synchronized method and release the object only if it was created. That also means you wouldn't need to pass `ObjectPool` to `AltsProtocolNegotiator` which assumed it is the same instance as `LazyChannel` is using.

Reply:
Good suggestion! done.